### PR TITLE
[DON'T MERGE] Test with Python 3.11

### PR DESCRIPTION
**Pull request by dstansby**
I saw https://discuss.python.org/t/the-cursed-release-of-python-3-11-0b4-is-now-available/17274 and thought it might be nice to do a single test with the Python 3.11 beta.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,131 +30,4 @@ jobs:
...
-    runs-on: ubuntu-latest
-    steps:
-      - uses: Cadair/matrix-notify-action@main
-        with:
-          matrix_token: ${{ secrets.matrix_access_token }}
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          homeserver: ${{ secrets.matrix_homeserver }}
-          roomid: '!JYqfIVJjWANcHnfktY:cadair.com'
-          ignore_pattern: '.*Load.*'
+        - linux: 3.11-dev
```
Comment (ConorMacBride) [inline .github/workflows/ci.yml:33]:
```suggestion
        - linux: py311
          python-version: '3.11-dev'
```
That should work (https://github.com/OpenAstronomy/github-actions-workflows/issues/53)

**Comment by ConorMacBride** [comment]
If you merge the following config into `tox.ini`, the alpha version of cython should install which should hopefully make more of these packages build successfully:

```yaml
[testenv]
pip_pre = true
```

**Comment by ConorMacBride** [comment]
Actually, these bug fixes seem to have been backported and are in the current cython release. The issue seems to be packages (including reproject) have pinned cython.

**Comment by Cadair** [comment]
Can we run a test build without some of the more tricky deps (looking square at you h5py / skimage)?

(I will go put my reproject hat on and fix that soon)

**Comment by Cadair** [comment]
Also is it possible to run a generic python pre-release build?

**Pull request closed without merging**